circle.yml

```diff
@@ -504,11 +504,11 @@ workflows:
       - cleanup
 
   # used for running cleanup on-demand
-  # by pushing to `ci-cleanup` branch
+  # by pushing to branches that start with `ci-cleanup/`
   ondemand-cleanup:
     jobs:
       - cleanup:
           filters:
             branches:
               only:
-                - ci-cleanup
+                - /ci-cleanup\/.*/
```

Comment:
@kewne, Can you provide an example of when this would be useful? I think sticking with a single branch here is a better idea. In that way, we can restrict pushes to `ci-cleanup` branch if needed. But changes of this commit mean, any push to `ci-cleanup/*` branches will run the job.

Reply:
@shimulch The idea is that this allows you to create "feature" branches that run the cleanup job, which is useful for reviewing because it shows directly in the PR, instead of having to `git push -f origin HEAD:ci-cleanup` and having to grab the build link every time you add a commit.
IMO having multiple branches isn't an issue because anyone with push permissions can always add their specific branch to the `.circle.yml` file anyway (which would have to be reverted before merging to `master`).

Ideally, there would be a way to add metadata to branches (or commits) that tells CircleCI that we want to run the cleanup job but I haven't found anything suitable; I'll gladly try other suggestions.

PS: what if I remove the `only:` part and make the deployment manual?

Reply:
@kewne, Yes, I thin removing `only` and making the deployment manual might be a better approach. Then we would not need to push empty commits to run a clean-up.

https://circleci.com/docs/2.0/workflows/#holding-a-workflow-for-a-manual-approval

Reply:
@shimulch 👍 Since the specific branch name is no longer necessary, I'm closing this in favor of the original #815